### PR TITLE
Implement a global reset and trim test suite

### LINKED ISSUE #1119: Minor refactor of __init__ and reset
Often we set and clear `Player` attributes in our `__init__` and `reset` methods, resulting in duplicate code and sometimes subtle bugs when we miss something in the `reset` method (or leave it out altogether).

I think it makes sense to have either a special method called something like `set_default_attributes` that handle all the attribute initialization and clearing that is called by the default `reset` method, or that we alter `Player.__init__` to call `reset` by default and set all the variables in the reset method. I prefer the former.

Thoughts?

### PR DESCRIPTION
**Pull request by drvinceknight**
This makes use of the `init_kwargs` and in fact we no longer need to
write reset methods (unless we want them for cheating strategies etc).

There are two relevant (simple) changes:

- Modify `player.reset` to call `self.__init__(**self.kwargs)`
- Tweak the match to pass the match attributes to the player just after resetting it.

The rest are me just removing the `reset` method except for the `Darwin` strategy which does things it's own way (and is classified as such).

Closes #1119

### CODE REVIEW

**Comment by drvinceknight** [comment]
This PR was having numerous CI failures due to hypothesis timeouts so I've gone through and trimmed a number of tests:

- Reduce examples of hypothesis
- Reduce some of the sizes of the larger integration tests
- Set defaults for some property based tests to only use short run time
  strategies

This now closes #1114

**Comment by drvinceknight** [comment]
There are two things going on in this PR. Usually we'd want to avoid that but in this case the first thing (adding the `reset`) was proving to be a good thing to "fix" as far as the tests go. The commits are tidy so if helpful:

- a04cfb6 implements the new reset method
- efefd13 trims our tests (less examples in hypothesis, use of short run time strategies etc) and tweaks the travis config to allow for a longer run time

There are two further commits:

- 4157070 adds an error just to make sure travis reports this correctly;
- 553e131 reverts 4157070

File: .travis.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ script:
   - cd docs; make html
   # Run the test suit with coverage
   - cd ..
-  - coverage run --source=axelrod -m unittest discover
+  - travis_wait 30 coverage run --source=axelrod -m unittest discover
```
Comment (marcharper) [inline .travis.yml]: Why is this needed?

**Comment by drvinceknight** [comment]
@marcharper and @meatballs you've both approved so I'm going to assume this is ok to go in?

File: .travis.yml (hunk 1 above)
Comment (drvinceknight) [inline .travis.yml]: If travis receives no output for a default 10 minutes then it stops and returns a failure. This changes the default to 30 mins but also changes how it runs it (https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received) - that's why I tested that failures were still being reported correctly.

Even with the test trimming the 10 minutes default was still causing failures on travis (I think due to travis being a bit slow and some hypothesis tests - I ran numerous experiments and wasn't getting this locally) so this just ensures that it shouldn't happen.
